File: frontend/lib/models/leave_type_model.dart
class LeaveType {
  final String id;
  final String name;
  final String code; // 'CASUAL', 'SICK', 'CUSTOM'

  LeaveType({
    required this.id,
    required this.name,
    required this.code,
  });

  factory LeaveType.fromJson(Map<String, dynamic> json) {
    return LeaveType(
      id: json['_id'] ?? json['id'] ?? '',
      name: json['name'] ?? '',
      code: json['code'] ?? 'CUSTOM',
    );
  }
}

File: backend/models/leaveTypeModel.js
// models/leaveTypeModel.js
import mongoose from 'mongoose';

const leaveTypeSchema = new mongoose.Schema(
  {
    company: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'Company',
      required: true,
      index: true,
    },
    // Display name shown in UI, e.g. "Casual Leave", "Sick Leave", "Birthday Leave"
    name: {
      type: String,
      required: true,
      trim: true,
    },
    // Marks system defaults
    code: {
      type: String,
      enum: ['CASUAL', 'SICK', 'CUSTOM'],
      default: 'CUSTOM',
    },
    isActive: {
      type: Boolean,
      default: true,
    },
  },
  { timestamps: true }
);

// Ensure a company cannot have two leave types with same name
leaveTypeSchema.index({ company: 1, name: 1 }, { unique: true });

const LeaveType = mongoose.model('LeaveType', leaveTypeSchema);
export default LeaveType;
